mm2src/coins/lightning.rs

```diff
@@ -647,38 +826,134 @@ pub async fn open_channel(ctx: MmArc, req: OpenChannelRequest) -> OpenChannelRes
         user_config.own_channel_config.our_htlc_minimum_msat = min;
     }
 
+    let rpc_channel_id = ln_coin.persister.get_last_channel_rpc_id().await? as u64 + 1;
+
     let temp_channel_id = async_blocking(move || {
         channel_manager
-            .create_channel(node_pubkey, amount_in_sat, push_msat, 1, Some(user_config))
+            .create_channel(node_pubkey, amount_in_sat, push_msat, rpc_channel_id, Some(user_config))
             .map_to_mm(|e| OpenChannelError::FailureToOpenChannel(node_pubkey.to_string(), format!("{:?}", e)))
     })
     .await?;
 
     {
-        let mut unsigned_funding_txs = ln_coin.platform_fields.unsigned_funding_txs.lock();
-        unsigned_funding_txs.insert(temp_channel_id, unsigned);
+        let mut unsigned_funding_txs = ln_coin.platform.unsigned_funding_txs.lock();
+        unsigned_funding_txs.insert(rpc_channel_id, unsigned);
     }
 
+    let pending_channel_details = SqlChannelDetails::new(
+        rpc_channel_id,
+        temp_channel_id,
+        node_pubkey,
+        true,
+        user_config.channel_options.announced_channel,
+    );
+
     // Saving node data to reconnect to it on restart
-    ln_coin.nodes_addresses.lock().insert(node_pubkey, node_addr);
-    ln_coin.persister.save_nodes_addresses(ln_coin.nodes_addresses).await?;
+    ln_coin.open_channels_nodes.lock().insert(node_pubkey, node_addr);
+    ln_coin
+        .persister
+        .save_nodes_addresses(ln_coin.open_channels_nodes)
+        .await?;
+
+    ln_coin.persister.add_channel_to_db(pending_channel_details).await?;
 
     Ok(OpenChannelResponse {
-        temporary_channel_id: temp_channel_id.into(),
+        rpc_channel_id,
         node_address: req.node_address,
     })
 }
 
 #[derive(Deserialize)]
-pub struct ListChannelsRequest {
+pub struct OpenChannelsFilter {
+    pub channel_id: Option<H256Json>,
+    pub counterparty_node_id: Option<PublicKeyForRPC>,
+    pub funding_tx: Option<H256Json>,
+    pub from_funding_value_sats: Option<u64>,
+    pub to_funding_value_sats: Option<u64>,
+    pub is_outbound: Option<bool>,
+    pub from_balance_msat: Option<u64>,
+    pub to_balance_msat: Option<u64>,
+    pub from_outbound_capacity_msat: Option<u64>,
+    pub to_outbound_capacity_msat: Option<u64>,
```

Comment:
GUIs and CLI users typically operate with `BigDecimal`. We prefer them to avoid performing number conversions their selves.

Reply:
`msat` has no decimals, or are you suggesting using sats instead of `msats` here? I avoided using BTC intentionally because these values can be really small and it would not look good presenting them in BTC.
I can see that mBTC (Milli-bitcoin) is used a lot in lightning GUIs, should we wait for the GUI design to know what's the best way to represent such amounts?

Reply:
Yes, we can postpone this decision because it's quite easy to change.